main/acle.md

```diff
@@ -1854,6 +1856,142 @@ or compiler-generated functions.
 
 This attribute does not apply to AArch64.
 
+## Function Multi Versioning
+
+The specification for Function Multi Versioning is in `BETA` state and may
+change or be extended in the future.
+
+Function Multi Versioning provides a convenient way to select the most
+appropriate version of a function at runtime. All versions of the function may
+be in the final binary. The compiler generates all supported versions and the
+runtime makes the selection at load time.
+
+The following attributes trigger the multi variant code generation:
+`target_version`, `target_clones`. These attributes can't be mixed. The
+"default" variant means the version of the function that would be generated
+without these attributes.
+
+This feature may be turned off in compile time. In this case the "default"
+variant shall be used.
+
+`__ARM_FEATURE_FUNCTION_MULTI_VERSIONING` is defined to 1 if available.
+
+``` c
+    __attribute__((target_version("name")))
+```
+
+when applied to a function it will become one of the versions. Function with the
+same name may exist with multiple versions in the same translation unit. One
+"default" version of the  function is required to be provided. Implicitly, when
+without attribute, or explicitly providing the "default" in the attribute. If
+only the "default" variant exist it should be linked directly.
+
+``` c 
+    __attribute__((target_clones("name",...)))
+```
+
+when applied to a function the compiler emits multiple versions based on the
+arguments. One of them is implicitly the "default". If a name is not recognized
+the compiler should ignore it. If the default matches with an other explicitly
+provided version the compiler can emit only one function instead of the two.
+
+Dependent features names could be joined by the `+` sign.
+
+None of these attributes will enable the corresponding ACLE features.
+
+These attributes have no effect on the calling convention, all versions must
+use the same calling convertion.
+
+### Name mangling
+
+The default version is not mangled top of the normal mangling.
+
+Version to be mangled. `<vendor-specific suffix>` should hold the version
+information [[cxxabi]](#cxxabi) as:
+
+``` c
+    _(M<name>)*
+```
+name is from table below.
+
+If multiple features are requested then those shall be applied in priority order.
```

Comment:
```suggestion
If multiple features are requested then those shall be applied following rules specified in section [Selection](#selection).
```

Reply:
Here I referred to the mangling, I'll add an example.